installer/util/brave_vpn_helper_utils.cc

```diff
@@ -0,0 +1,270 @@
+      .Append(install_static::kCompanyPathName)
+      .Append(brave_vpn::GetBraveVpnHelperServiceName());
+}
+
+bool InstallBraveVPNHelperService() {
+  base::CommandLine service_cmd(GetVpnHelperServiceProfileDir());
+  installer::InstallServiceWorkItem install_service_work_item(
+      brave_vpn::GetBraveVpnHelperServiceName(),
+      brave_vpn::GetBraveVpnHelperServiceDisplayName(), SERVICE_DEMAND_START,
+      service_cmd, base::CommandLine(base::CommandLine::NO_PROGRAM),
+      brave_vpn::kBraveVpnHelperRegistryStoragePath, {}, {});
+  install_service_work_item.set_best_effort(true);
+  install_service_work_item.set_rollback_enabled(false);
+  if (install_service_work_item.Do()) {
+    auto success =
+        ConfigureServiceAutoRestart(brave_vpn::GetBraveVpnHelperServiceName(),
+                                    brave_vpn::GetBraveVPNConnectionName());
+    return success;
+  }
+  return false;
+}
+
+bool IsBraveVPNHelperServiceInstalled() {
+  ScopedScHandle scm(::OpenSCManager(nullptr, nullptr, SC_MANAGER_CONNECT));
+  if (!scm.IsValid()) {
+    VLOG(1) << "::OpenSCManager failed. service_name: "
+            << brave_vpn::GetBraveVpnHelperServiceName()
+            << ", error: " << std::hex << HRESULTFromLastError();
+    return false;
+  }
+  ScopedScHandle service(::OpenService(
+      scm.Get(), brave_vpn::GetBraveVpnHelperServiceName().c_str(),
+      SERVICE_QUERY_STATUS));
+
+  // Service registered and has not exceeded the number of auto-configured
+  // restarts.
+  return service.IsValid();
+}
+
+bool IsNetworkFiltersInstalled() {
+  DCHECK(IsBraveVPNHelperServiceInstalled());
+  base::win::RegKey service_storage_key(
+      HKEY_LOCAL_MACHINE, brave_vpn::kBraveVpnHelperRegistryStoragePath,
+      KEY_READ);
+  if (!service_storage_key.Valid()) {
+    return false;
+  }
+  DWORD current = -1;
+  if (service_storage_key.ReadValueDW(
+          brave_vpn::kBraveVpnHelperFiltersInstalledValue, &current) !=
+      ERROR_SUCCESS) {
+    return false;
+  }
+  return current > 0;
+}
+
+HRESULT InstallBraveVPNHelperServiceImpersonated() {
+  HRESULT hr = ::CoImpersonateClient();
+  if (FAILED(hr)) {
+    return hr;
+  }
+
+  base::ScopedClosureRunner revert_to_self(
+      base::BindOnce([]() { ::CoRevertToSelf(); }));
+
+  auto executable_path = brave_vpn::GetBraveVpnHelperServicePath();
+  base::CommandLine command_line(executable_path);
+  command_line.AppendSwitch(brave_vpn::kBraveVpnHelperInstall);
+
+  base::LaunchOptions options = base::LaunchOptions();
+  options.feedback_cursor_off = true;
+  options.wait = true;
+  base::Process proc = base::LaunchProcess(command_line, options);
+  if (!proc.IsValid()) {
+    return brave_vpn::HRESULTFromLastError();
+  }
+
+  return S_OK;
+}
+
+HRESULT InstallBraveWireGuardServiceImpersonated() {
+  HRESULT hr = ::CoImpersonateClient();
+  if (FAILED(hr)) {
+    return hr;
+  }
+
+  base::ScopedClosureRunner revert_to_self(
+      base::BindOnce([]() { ::CoRevertToSelf(); }));
```

Comment:
Nit. Maybe? 

```c++
absl::Cleanup cleanup([]() { ::CoRevertToSelf(); });
```

Not important, but `ScopedClosureRunner` incurs in heap allocations.